src/shared/proto/robot_component_status.proto

```diff
@@ -0,0 +1,139 @@
+syntax = "proto3";
+
+/* Data about the status of the break beam */
+message BreakBeamStatus
+{
+    // Whether or not the ball is interrupting the robot’s laser beam.
+    bool ball_in_beam = 1;
+
+    // The reading of the robot’s laser sensor, normalized to [0,1]
+    float break_beam_reading = 2;
+}
+
+/* Data about the status of the chipper and kicker */
+message ChipperKickerStatus
+{
+    // Indicates the time elapsed since chipper was last fired in ms
+    // UINT64_MAX indicates chipper was never fired
+    uint64 time_since_chipper_fired = 1;
+
+    // Indicates the time elapsed since kicker was last fired in ms
+    // UINT64_MAX indicates kicker was never fired
+    uint64 time_since_kicker_fired = 2;
+}
+
+/* Data about the status of the dribbler
+ * Note that the rotation is from the perspective of rear of the motor, looking down
+ * the shaft starting from the motor body
+ */
+message DribblerStatus
+{
+    float dribbler_rpm = 1;
+}
+
+/* Error codes are to be displayed in the visualizer */
+enum ErrorCode
+{
+    // Default message required for proto enums
+    NO_ERROR = 0;
+
+    // Level triggered events
+    LOW_CAP                         = 1;
+    LOW_BATTERY                     = 2;
+    HIGH_BOARD_TEMP                 = 3;
+    CHARGE_TIMEOUT                  = 4;
+    WHEEL_0_MOTOR_HOT               = 5;
+    WHEEL_1_MOTOR_HOT               = 6;
+    WHEEL_2_MOTOR_HOT               = 7;
+    WHEEL_3_MOTOR_HOT               = 8;
+    DRIBBLER_MOTOR_HOT              = 9;
+    WHEEL_0_ENCODER_NOT_COMMUTATING = 10;
+    WHEEL_1_ENCODER_NOT_COMMUTATING = 11;
+    WHEEL_2_ENCODER_NOT_COMMUTATING = 12;
+    WHEEL_3_ENCODER_NOT_COMMUTATING = 13;
+    WHEEL_0_HALL_SENSOR_STUCK_LOW   = 14;
+    WHEEL_1_HALL_SENSOR_STUCK_LOW   = 15;
+    WHEEL_2_HALL_SENSOR_STUCK_LOW   = 16;
+    WHEEL_3_HALL_SENSOR_STUCK_LOW   = 17;
+    DRIBBLER_HALL_SENSOR_STUCK_LOW  = 18;
+    WHEEL_0_HALL_SENSOR_STUCK_HIGH  = 19;
+    WHEEL_1_HALL_SENSOR_STUCK_HIGH  = 20;
+    WHEEL_2_HALL_SENSOR_STUCK_HIGH  = 21;
+    WHEEL_3_HALL_SENSOR_STUCK_HIGH  = 22;
+    DRIBBLER_HALL_SENSOR_STUCK_HIGH = 23;
+
+    // Edge triggered events
+    ICB_CRC_ERROR                        = 24;
+    RECEIVE_FRAME_CHECK_SEQUENCE_FAILURE = 25;
+    CRASHED_CORE_DUMPED                  = 26;
+    CRASHED_NO_CORE_DUMP                 = 27;
+
+    // Radio Dongle Errors
+    ROBOT_DEAD          = 28;
+    ESTOP_BROKEN        = 29;
+    RX_FCS_FAIL         = 30;
+    SECOND_DONGLE       = 31;
+    TRANSMIT_QUEUE_FULL = 32;
+    RECEIVE_QUEUE_FULL  = 33;
+}
+
+/* Firmware build information  */
+message FirmwareStatus
+{
+    // The microcontroller firmware build ID.
+    uint32 fw_build_id = 1;
+}
+
+/* Encoder data from a single motor
+ * Note that the rotation is from the perspective of rear of the motor, looking down
+ * the shaft starting from the motor body
+ */
+message DriveUnit
+{
+    float wheel_rpm          = 1;
```

Comment:
Very nit:

Can we get wheel_ang_vel (in rad/s) instead of RPM? It's just more usable and I see RPM being converted anyways.